controllers/openstackcluster_controller.go

```diff
@@ -448,7 +448,7 @@ func bastionToInstanceSpec(openStackCluster *infrav1.OpenStackCluster, clusterNa
 	}
 
 	instanceSpec.SecurityGroups = openStackCluster.Spec.Bastion.Instance.SecurityGroups
-	if openStackCluster.Spec.ManagedSecurityGroups {
+	if openStackCluster.Spec.ManagedSecurityGroups.Enabled {
```

Comment:
If we changed this into a struct pointer, I think all of these would just become nil checks. That looks fairly safe as far as pointers go, given the current usage.